### PR TITLE
docs(examples): adds java spring boot examples

### LINKED ISSUE #5190: Java Web SDK and Example/Guide
**Acceptance Criteria**
- [ ] We provide an authentication package extending spring security
     - [ ] Authentication is implemented with OIDC Standard and Authorization Code Flow including PKCE
     - [ ] The session of an authenticated users is persistent (e.g session cookie, probably provided by spring security)
     - [ ] The user profile can be loaded from the userinfo_endpoint
     - [ ] It is possible to check if a user has a specific role (role can be taken from userinfo endpoint)
     - [ ] Logout is implemented, by using the end_session endpoint and local session must be deleted as well
- [ ] The usage is shown in an example, which included the following:
     - [x] Example page, which redirects user to login if not authenticated
     - [x] Example page, when user is authenticated, profile information is shown (loaded from userinfo_endpoint)
     - [x] Example admin page, if user has a role admin, ~an additional page can be shown with the text "this is an administrator page"~ the role is displayed (and on the tasks example, the user is able to add new tasks)
     - [x] In the example app I can press a logout button, after that the user can't access the profile page and has to re-authenticate.
- [x] We provide a guide in the ZITADEL docs, which shows how to use/implement analog the angular guide: https://zitadel.com/docs/examples/login/angular
- [ ] Readme of the package includes the following:
  - [x] Features
  - [x] Link to documentation page and guide
  - [ ] Package structure
  - [x] How to run examples
- [ ] Package is published on a registry 

**Additional Information**
- We already do have a java sdk, we should extend/update that one https://github.com/zitadel/zitadel-java
- Versions should be up to date afterwards and also on the newest standards regarding implementation

More specific description here: #5124

### PR DESCRIPTION
**Pull request by livio-a**
closes #5190 and #5191 

- [x] requires branch reference update after https://github.com/zitadel/zitadel-java/pull/2 is merged

### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Comment by livio-a** [comment]
@muhlemmer i've just added code highlighting for java (and php 😉 )

File: docs/docs/examples/secure-api/java-spring.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+sidebar_label: Java Spring Boot
+---
+
+This integration guide shows you how to integrate **ZITADEL** into your Java Spring Boot API. It demonstrates how to secure your API using
+OAuth 2 Token Introspection.
+
+At the end of the guide you should have an API with a protected endpoint.
+
+> This documentation references our [example](https://github.com/zitadel/zitadel-java) on GitHub.
+> You can either create your own application or directly run the example by providing the necessary arguments.
```
Comment (muhlemmer) [inline docs/docs/examples/secure-api/java-spring.md:12]:
```suggestion
:::info
This documentation references our [example](https://github.com/zitadel/zitadel-java) on GitHub.
You can either create your own application or directly run the example by providing the necessary arguments.
:::
```

File: docs/docs/examples/secure-api/java-spring.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+This guide will reference the [example repository](https://github.com/zitadel/zitadel-java) and explain the necessary steps taken in there.
+If your starting from scratch, you can use the Spring Initializer with the [following setup](https://start.spring.io/#!type=maven-project&language=java&platformVersion=3.2.1&packaging=jar&jvmVersion=17&dependencies=web,lombok,oauth2-resource-server) as a base.
+
+### Support class
+
+To be able to take the most out of ZITADELs RBAC, we first need to create a CustomAuthorityOpaqueTokenIntrospector, that will
+customize the introspection behaviour and map the role claims (`urn:zitadel:iam:org:project:roles`)
+into Spring Security `authiorities`, which can be used later on to determine the granted permissions.
+
+So in your application, create a 'support/zitadel' package and in there the `CustomAuthorityOpaqueTokenIntrospector.java`:
```
Comment (muhlemmer) [inline docs/docs/examples/secure-api/java-spring.md:42]:
```suggestion
So in your application, create a `support/zitadel` package and in there the `CustomAuthorityOpaqueTokenIntrospector.java`:
```

File: docs/docs/examples/login/java-spring.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+sidebar_label: Java Spring Boot
+---
+
+This integration guide demonstrates the recommended way to incorporate ZITADEL into your Spring Boot web application. 
+It explains how to enable user login in your application and how to fetch data from the user info endpoint.
+
+By the end of this guide, your application will have login functionality and will be able to access the current user's profile.
+
+> This documentation references our [example](https://github.com/zitadel/zitadel-java) on GitHub. 
+> You can either create your own application or directly run the example by providing the necessary arguments.
```
Comment (muhlemmer) [inline docs/docs/examples/login/java-spring.md:12]:
```suggestion
:::info
This documentation references our [example](https://github.com/zitadel/zitadel-java) on GitHub. 
You can either create your own application or directly run the example by providing the necessary arguments.
:::
```

File: docs/docs/examples/login/java-spring.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+We recommend that you use [Proof Key for Code Exchange (PKCE)](/apis/openidoauth/grant-types#proof-key-for-code-exchange) for all applications.
+
+![Create app in console - set auth method](/img/java-spring/app-create-auth.png)
+
+### Redirect URIs
+
+The Redirect URIs field tells ZITADEL where it's allowed to redirect users after authentication. For development, you can set dev mode to `true` to enable insecure HTTP and redirect to a `localhost` URI.
+The Post-logout redirect send the users back to a route on your application after they have logged out.
+
+> If you are following along with the [example](https://github.com/zitadel/zitadel-java), set the dev mode to `true`, the Redirect URIs to `http://localhost:18080/webapp/login/oauth2/code/zitadel` and Post redirect URI to `http://localhost:18080/webapp`.
```
Comment (muhlemmer) [inline docs/docs/examples/login/java-spring.md:31]:
```suggestion
:::info
If you are following along with the [example](https://github.com/zitadel/zitadel-java), set the dev mode to `true`, the Redirect URIs to `http://localhost:18080/webapp/login/oauth2/code/zitadel` and Post redirect URI to `http://localhost:18080/webapp`.
:::
```

**Review by muhlemmer — CHANGES_REQUESTED** [review]

**Review by muhlemmer — CHANGES_REQUESTED** [review]
Some formatting and info boxes.

**Review by muhlemmer — APPROVED** [review]

**Comment by livio-a** [comment]
i've updated the code references (to main and removed the java/spring-boot folder)

**Pull request merged**